ci/prepare-unity.sh

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env bash
+set -e -u -x -o pipefail
+
+cd "$(dirname "$0")/../"
+
+source ci/includes/pinned-tools.sh
+source ci/includes/profiling.sh
+
+function isLinux() {
+  [[ "$(uname -s)" == "Linux" ]];
+}
+
+function isMacOS() {
+  [[ "$(uname -s)" == "Darwin" ]];
+}
+
+function isWindows() {
+  ! ( isLinux || isMacOS );
+}
+
+# Only run inside TeamCity
+if [ -z "${TEAMCITY_CAPTURE_ENV+x}" ]; then
+    exit 0
+fi
+
+if isWindows; then
+    MODULES="linux mac-mono"
+elif isMacOS; then
+    MODULES="linux windows"
+elif isLinux; then
+    echo "Building is not supported on linux"
+    exit 1
+fi
+
+GOPATH="$(pwd)/go"
+export GOPATH
+UNITY_PACKAGE="github.com/improbable/unity_downloader"
+
+markStartOfBlock "$0"
+
+# Check if version is installed
+if isUnityImprobablePathPresent; then
```

Comment:
this is kind of confusing as it checks for `UNITY_DIR` which could be our internal path that we use for TC agents: `C:/Unity/..` or it could be that `UNITY_HOME` is set to some arbitrary value. maybe rename it or move this check `[ -d "$UNITY_DIR" ]` directly into the if statement

also: add a log message here as well. We need to tell people what's going on, if we ever need to debug the logs ;)

Reply:
I'll add a log that unity is already installed if this passes.
I do not understand the issue with the path check?

Reply:
what I meant by that is that you changed the implementation of `isUnityImprobablePathPresent`. Previously it checked whether Unity exists on the path for our TC agents. Now it checks whether it exists in `UNITY_DIR`, which could be either the "Improbable path" or the "User path". I guess renaming the method would be sufficient. 

Though with Jared's PR that needs to change anyway. so maybe directly do the `[ -d "$UNITY_DIR" ]` check here instead of calling `isUnityImprobablePathPresent`